pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -604,9 +604,10 @@ func (os *OpenStack) LoadBalancer() (cloudprovider.LoadBalancer, bool) {
 	return &LbaasV2{LoadBalancer{network, compute, lb, os.lbOpts}}, true
 }
 
+// this tries to catch 404 error from gophercloud
 func isNotFound(err error) bool {
-	e, ok := err.(*gophercloud.ErrUnexpectedResponseCode)
-	return ok && e.Actual == http.StatusNotFound
+	httpStatus, ok := err.(gophercloud.ErrDefault404)
```

Comment:
How have you tested this?   From a quick search of the code, it seems `gophercloud.ErrResourceNotFound` is also used throughout the gophercloud codebase (for cases when the expected resource wasn't found in a list operation, rather than an actual 404 HTTP response).

I think we want this function to be:
```go
func isNotFound(err error) bool {
    switch err.(type) {
    case gophercloud.ErrDefault404: return true
    case gophercloud.ErrResourceNotFound: return true
    }
    return err == ErrNotFound
}
```

.. and tests! It's embarrassing that we merged an update that broke this function without noticing..

Reply:
related ... please look at this too - https://github.com/kubernetes/kubernetes/pull/61002/commits/0da1660653b82d433cd92c6ba2b79df5d51b7bf8

Reply:
I think we should use #61002 version to isnotfound function. However, I could still fix that waitLoadbalancerDeleted function timeout thing in this PR